service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceTwin.java

```diff
@@ -538,4 +659,21 @@ private DeviceTwinDevice jsonToDeviceTwinDevice(String json) throws IOException
 
         return deviceTwinDevice;
     }
+
+    private String getAuthenticationToken()
+    {
+        // Three different constructor types for this class, and each type provides either a TokenCredential implementation,
+        // an AzureSasCredential instance, or just the connection string. The sas token can be retrieved from the non-null
+        // one of the three options.
+        if (this.credential != null)
+        {
+            return this.credential.getToken(new TokenRequestContext()).block().getToken();
```

Comment:
Confirm with service team on what prefix and suffix to add.
https://github.com/Azure/azure-sdk-for-net/blob/8970711b5c82c9fac3f7135529ab57de5a294f08/sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs#L218